renormalizer/mps/gs.py

```diff
@@ -366,11 +366,11 @@ def get_ham_direct(
 def sign_fix(c, nroots):
     if nroots > 1:
         if isinstance(c, list):
-            return [ci / np.sign(np.max(np.abs(ci))) for ci in c]
+            return [ci / np.sign(ci[np.abs(ci).argmax()]) for ci in c]
         else:
-            return c / np.sign(np.max(c, axis=0))
+            return c / np.sign(np.max(np.abs(c), axis=0))
```

Comment:
It seems in this branch it still does not work properly?

Reply:
thanks for pointing out! Now I think it is resolved.